### PR TITLE
[CSL-1563] Move 'gitRev' out of 'cardano-sl'

### PR DESCRIPTION
**Pull request by rockbmb**

### CODE REVIEW

File: core/Pos/Core/Constants/Raw.hs [hunk 1]
```diff
@@ -200,6 +201,15 @@ isDevelopment = True
 isDevelopment = False
 #endif
 
+#define QUOTED(x) "/**/x/**/"
+
+gitRev :: Text
```
Comment (volhovM) [inline core/Pos/Core/Constants/Raw.hs]: How is this going to work? You've moved the function that depends on a environment variable from `node` to `core` which doesn't solve the problem: changing git revision outside will change it and lead to recompilation of core and all dependent modules.

File: scripts/build/cardano-sl.sh [hunk 2]
```diff
@@ -327,19 +327,25 @@ echo "'explorer' flag: $explorer"
...
+  sbuild="stack build --ghc-options=\"$ghc_opts_2\" $commonargs $norun --dependencies-only $args $prj"
   echo -e "$sbuild\n"
   eval $sbuild
 
   if [[ $no_code == true ]]; then
-    ghc_opts_2="$ghc_opts -fwrite-interface -fno-code"
+    ghc_opts_3="$ghc_opts2 -fwrite-interface -fno-code"
   else
-    ghc_opts_2="$ghc_opts"
+    ghc_opts_3="$ghc_opts2"
```
Comment (volhovM) [inline scripts/build/cardano-sl.sh]: Why did you switch to `ght_opts_N`? I personally find it less convenient than overriding `ghc_opts`.

**Comment by gromakovsky** [comment]
@volhovM see [my comment](https://issues.serokell.io/issue/CSL-1563#comment=96-7017). Moving `gitRev` into executables won't solve the problem, because we will still need to pass `DGITREV=bla` to `cardano-sl` which will recompile whole `cardano-sl` whenever something changes. There may be a way to pass CPP definition only to executables, but I am not aware of it.
If we pass `DGITREV` only to `core`, `stack` will rebuild everything only when something changes in `core` which happens substantially less often than in all packages.
The problem with this solution is that `GITREV` may easily point to wrong commit. Note, however, that this problem exists anyway. In current master if you change revision but don't change anything in `cardano-sl`, it won't rebuild anything.
I don't know what is the ideal solution, we should probably ask in #haskell-best-practice channel.

**Comment by rockbmb** [comment]
@gromakovsky have we decided how to move forward with this?

**Comment by gromakovsky** [comment]
@rockbmb yes, see [here](https://issues.serokell.io/issue/CSL-1563#comment=96-7109).

**Comment by gromakovsky** [comment]
I also asked a question in #sl-devops: https://input-output-rnd.slack.com/archives/G3TLHGM18/p1504948435000038

**Comment by gromakovsky** [comment]
See #1647.

**Pull request closed without merging**